src/controller/java/AndroidCallbacks.cpp

```diff
@@ -281,6 +283,34 @@ void ReportCallback::OnAttributeData(const app::ConcreteDataAttributePath & aPat
     env->CallVoidMethod(mNodeStateObj, addAttributeMethod, static_cast<jint>(aPath.mEndpointId),
                         static_cast<jlong>(aPath.mClusterId), static_cast<jlong>(aPath.mAttributeId), attributeStateObj);
     VerifyOrReturnError(!env->ExceptionCheck(), env->ExceptionDescribe());
+
+    UpdateClusterDataVersion();
+}
+
+void ReportCallback::UpdateClusterDataVersion()
+{
+    JNIEnv * env = JniReferences::GetInstance().GetEnvForCurrentThread();
+    chip::app::ConcreteClusterPath lastConcreteClusterPath;
+    if (mClusterCacheAdapter.GetLastReportDataPath(lastConcreteClusterPath) == CHIP_NO_ERROR &&
```

Comment:
nit: consider using early return to decrease nesting, so that code looks more linear. Like:

```cpp
CHIP_ERROR err = mClusterCacheAdapter.GetLastReportDataPath(lastConcreteClusterPath);

if (err != CHIP_NO_ERROR) {
   // consider logging
   return;
}

if (!lastConcreteClusterPath.IsValidConcreteClusterPath()) {
   return; // document in what circumstances this happens
}

chip::Optional<chip::DataVersion> committedDataVersion;
err = mClusterCacheAdapter.GetVersion(lastConcreteClusterPath, committedDataVersion);
if (err != CHIP_NO_ERROR) {
    /// consider logging here
    return; 
}

if (!committedDataVersion.HasValue()) {
    return; // document on when this happens. Is this normal? if not, maybe log?
}

jmethodID setDataVersionMethod;
//..... rest of code
```

Reply:
done，thanks